src/trading_api_functions.jl

```diff
@@ -0,0 +1,541 @@
+symbol 	            string 	                    Required            symbol, asset ID, or currency pair to identify the asset to trade
+qty 	            string<number> 	            Required            number of shares to trade. Can be fractionable for only market and day order types.
+notional 	        string<number> 	            Required            dollar amount to trade. Cannot work with qty. Can only work for market order types and day for time in force.
+side 	            string 	                    Required            buy or sell
+type 	            string 	                    Required            market, limit, stop, stop_limit, or trailing_stop
+time_in_force 	    string 	                    Required            Please see Understand Orders for more info on what values are possible for what kind of orders. :: day, gtc, opg, cls, ioc, fok
+limit_price 	    string<number> 	            Required            required if type is limit or stop_limit
+stop_price 	        string<number> 	            Required            required if type is stop or stop_limit
+trail_price 	    string<number> 	            Required            this or trail_percent is required if type is trailing_stop
+trail_percent 	    string<number> 	            Required            this or trail_price is required if type is trailing_stop
+extended_hours 	    boolean 	                Optional            (default) false. If true, order will be eligible to execute in premarket/afterhours. Only works with type limit and time_in_force day.
+client_order_id 	string (<= 48 characters) 	Optional            A unique identifier for the order. Automatically generated if not sent.
+order_class 	    string 	                    Optional            simple, bracket, oco or oto. The empty string ("") is synonym for simple. For details of non-simple order classes, please see Bracket Order Overview
+take_profit 	    object 	                    Optional            Additional parameters for take-profit leg of advanced orders
+stop_loss 	        object 	                    Optional            Additional parameters for stop-loss leg of advanced orders
+
+# example 
+place_order("AAPL"; qty="1", notional=nothing, side="buy", type="limit", time_in_force="gtc", limit_price="150.56", stop_price=nothing,
+    trail_price=nothing, trail_percent=nothing, extended_hours=nothing, client_order_id=nothing, order_class=nothing, take_profit=nothing, stop_loss=nothing)
+
+
+place_order("AAPL"; qty="1", notional=nothing, side="sell", type="limit", time_in_force="day", limit_price="155.56", stop_price=nothing,
+    trail_price=nothing, trail_percent=nothing, extended_hours=nothing, client_order_id=nothing, order_class=nothing, take_profit=nothing, stop_loss=nothing)
+"""
+function place_order(symbol::String; qty::Any=nothing, notional::Any=nothing, side::Any=nothing, type::Any=nothing, time_in_force::Any=nothing, limit_price::Any=nothing, stop_price::Any=nothing,
```

Comment:
rather than having the if statements in the body, do we not want to split out placing a market/limit/stop order? 
This will clean up the code and make it more readable

Reply:
ok all the booleans here - I coded the input cases for a singular function vs letting the API ping back the errors. for example if select type as "market" and then you have a limit price - API will ping back an error. So all these @assert, assertations are constraining the input from the user that is between the required fields, then if select x field, other y fields become negative or others become required. We can either allow the control of it here or leave the function wide open and let the API itself throw the exceptions. Let me know your thoughts.

Reply:
Ok i see what you mean - there could be an an order type per available type.

` available arguments :: market, limit, stop, stop_limit, or trailing_stop`

this would limit all the assertations - would be just those to the type of order.